changelog/fragments/k8s-1.24-bump.yaml

```diff
@@ -0,0 +1,34 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+entries:
+  - description: >
+      Bump dependencies to support Kubernetes 1.24. Projects scaffolded with the go/v3 plugin now support Go 1.18.
+      Dependency bumps that occurred are as follows:
+      
+      github.com/operator-framework/api --> v0.15.0
+      github.com/operator-framework/helm-operator-plugins --> commit d3967d2ae2acf40cca1bf4215508b10a55ea6755
+      github.com/operator-framework/java-operator-plugins --> commit 95356a3d7a6581dba1eb3ce641b72a1f7b2e542a
+      github.com/operator-framework/operator-lib --> v0.11.0
+      github.com/operator-framework/operator-manifest-tools --> v0.2.1
+      github.com/operator-framework/operator-registry --> v1.23.0
+      helm.sh/helm/v3 --> v3.9.0
+      k8s.io/api --> v0.24.0
+      k8s.io/apiextensions-apiserver --> v0.24.0
+      k8s.io/apimachinery --> v0.24.0
+      k8s.io/cli-runtime --> v0.24.0
+      k8s.io/client-go --> v0.24.0
+      k8s.io/kubectl --> v0.24.0
+      sigs.k8s.io/controller-runtime --> v0.12.1
+      sigs.k8s.io/controller-tools --> v0.9.0
+      sigs.k8s.io/kubebuilder/v3 --> commit eea565cb3f508d7bcf10831d2b761dbf59af6db0
```

Comment:
we should only add what impacts the users in the scaffolds.
Ie. ` sigs.k8s.io/kubebuilder/v3 --> commit eea565cb3f508d7bcf10831d2b761dbf59af6db0` is irrelevant for those who uses SDK CLI the only get the latest release and migrate their project to comply with. their project does not use KB as a dep

Reply:
That commit does modify the scaffolding logic by bumping controller-tools to v0.9.0 though.

https://github.com/kubernetes-sigs/kubebuilder/commit/eea565cb3f508d7bcf10831d2b761dbf59af6db0

The first file i see in the commit is that the go/v3 scaffolding was modified

Reply:
then we need to have a new entry here and clarify that we upgrade controller-tools from to x and how users can do the same in project that are scaffold prior these change

c/c @everettraven

Reply:
I updated the changelog in https://github.com/operator-framework/operator-sdk/pull/5843/commits/b7b0924a3a2aef6416899e81f51d754e32f0ce2b

Let me know if the migration there looks good!

Reply:
Final changelog should be in commit https://github.com/operator-framework/operator-sdk/pull/5843/commits/b30c7a866b98ba3603f840af06110339c21aeb91